### PR TITLE
challenge[jsArray2_animalObjects]: Challenge Done

### PR DESCRIPTION
**Pull request by jamoragar**

### CODE REVIEW

File: javascript/js-array-methods-2_animal-objects/index.js [hunk 1]
```diff
@@ -91,22 +91,23 @@ const indexOfAnimalWithNameLongerFive = null;
 // Hint: sort() mutates the original array, which is bad.
 // -> Use animals.slice().sort(...) to make a copy (and the tests work).
 
-const animalsSortedAlphabetically = null;
+const animalsSortedAlphabetically = animals.slice().sort((a, b) => a.name < b.name ? -1 : 1);
 
-const animalsSortedByWeightStartingWithLowest = null;
+const animalsSortedByWeightStartingWithLowest = animals.slice().sort((a, b) => a.weight < b.weight ? -1 : 1);
```
Comment (mgczarnievicz) [inline javascript/js-array-methods-2_animal-objects/index.js]: When they are number, you can just return the value of `a-b`

File: javascript/js-array-methods-2_animal-objects/index.js [hunk 2]
```diff
@@ -91,22 +91,23 @@ const indexOfAnimalWithNameLongerFive = null;
 // Hint: sort() mutates the original array, which is bad.
 // -> Use animals.slice().sort(...) to make a copy (and the tests work).
 
-const animalsSortedAlphabetically = null;
+const animalsSortedAlphabetically = animals.slice().sort((a, b) => a.name < b.name ? -1 : 1);
```
Comment (mgczarnievicz) [inline javascript/js-array-methods-2_animal-objects/index.js]: be careful

File: javascript/js-array-methods-2_animal-objects/index.js [hunk 3]
```diff
@@ -91,22 +91,23 @@ const indexOfAnimalWithNameLongerFive = null;
...
 
-const animalsSortedByWeightStartingWithLowest = null;
+const animalsSortedByWeightStartingWithLowest = animals.slice().sort((a, b) => a.weight < b.weight ? -1 : 1);
 
-const animalsSortedByWeightReversed = null;
+const animalsSortedByWeightReversed = animals.slice().sort((a, b) => a.weight > b.weight ? -1 : 1);
```
Comment (mgczarnievicz) [inline javascript/js-array-methods-2_animal-objects/index.js]: you could use the `reverse` method.

File: javascript/js-array-methods-2_animal-objects/index.js (hunk 2 above)
Comment (jamoragar) [reply javascript/js-array-methods-2_animal-objects/index.js]: Yes! I have already refactored it

File: javascript/js-array-methods-2_animal-objects/index.js (hunk 1 above)
Comment (jamoragar) [reply javascript/js-array-methods-2_animal-objects/index.js]: Thanks for the tip 🫶🏼

**Pull request merged**